### PR TITLE
NOTIF-190 Separate old and new cronjob

### PR DESCRIPTION
**Pull request by bennetelli**

### CODE REVIEW

**Comment by bennetelli** [comment]
I separated the old cron job from the new Kafka input way.

The class was too huge IMO and with this separation it's easier to understand and easier to remove the old way later.

File: backend/src/main/java/com/redhat/cloud/notifications/processors/email/OldDailyEmailCronJob.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    EmailAggregationResources emailAggregationResources;
+
+    @Inject
+    EmailSender emailSender;
+
+    @Inject
+    EmailSubscriptionTypeProcessor emailSubscriptionTypeProcessor;
+
+    @Override
+    public Multi<NotificationHistory> process(Action action, List<Endpoint> endpoints) {
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/processors/email/OldDailyEmailCronJob.java:64]: This methods processes `instant` emails also, it should remain in `EmailSubscriptionTypeProcessor`.

File: backend/src/main/java/com/redhat/cloud/notifications/processors/email/OldDailyEmailCronJob.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        return processUni.onItem().transformToMulti(_unused -> sendInstantEmail(
+                action,
+                Set.copyOf(endpoints),
+                template
+        ));
+    }
+
+    private Multi<NotificationHistory> sendInstantEmail(Action action, Set<Endpoint> endpoints, EmailTemplate emailTemplate) {
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/processors/email/OldDailyEmailCronJob.java:98]: `instant email related`

File: backend/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java [hunk 3]
```diff
@@ -3,61 +3,41 @@
...
 import java.util.Map;
-import java.util.Set;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import static java.time.ZoneOffset.UTC;
+import static com.redhat.cloud.notifications.models.EmailSubscriptionType.*;
 
 @ApplicationScoped
-public class EmailSubscriptionTypeProcessor implements EndpointTypeProcessor {
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java:43]: We still need to implement `EndpintTypeProcessor`

**Review by josejulio — CHANGES_REQUESTED** [review]
I like the idea to group the soon-to-remove code, but these changes are also moving the mechanism to process instant emails.

**Comment by bennetelli** [comment]
Will leave it as it is right now.

**Pull request closed without merging**